pages/documents/DeveloperTools/FaaS/Developing/best-pratices.md

```diff
@@ -0,0 +1,241 @@
+  }
+}
+```
+
+### OAuth 2.0
+
+LivePerson Functions supports the Client Credentials as `Grant Type`. The [Client Credentials](https://oauth.net/2/grant-types/client-credentials/) grant type is used by clients to obtain an access token outside of the context of a user. 
+
+<div class="important">
+  It's recommended to store the <code>clientId</code> and <code>clientSecret</code> in the <a href="liveperson-functions-development-toolbelt.html#secret-storage-client">secret storage</a>
+</div>
+
+```javascript
+async function lambda(input, callback) {
+  const { Toolbelt } = require("lp-faas-toolbelt");
+  const httpClient = Toolbelt.HTTPClient(); 
+  const secretClient = Toolbelt.SecretClient();
+
+  try {
+    const authUrl = 'www.mywebsite.com/auth';
+    const URL = 'www.mywebsite.com/request';
+
+    // Get clientId and clientSecret from secret storage
+    const client_id = await secretClient.readSecret('clientId');
+    const client_secret = await secretClient.readSecret('clientSecret');
+
+    // Obtain the token from the authorization server
+    const token = await hpptClient(authUrl, {
+        method: 'POST',
+        form: {
+            client_id: client_id.value,
+            client_secret: client_secret.value,
+            grant_type: 'client_credentials'
+        },
+        json: true,
+        simple: true,
+        resolveWithFullResponse: false
+    })
+
+    // Perform request with token (type may change)
+    const response = await httpClient(URL, {
+      method: "GET", 
+      headers: {
+        'Authorization': `Bearer ${token}`,
+      },
+      simple: false,
+      resolveWithFullResponse: true
+    });
+    
+    callback(null, token);
+  } catch (err) {
+    callback(err, null)
+  }
+}
+```
+
+### Usage of Environment Variables
+
+The Environment Variables are used to save configurations for a specific lambda. With this approach you can separate static variables from the code.
+
+<div class="important">
+  Do not save sensitive data in Environment Variables!
+</div>
+
+```javascript
+async function lambda(input, callback) {
+  const { Toolbelt } = require("lp-faas-toolbelt");
+  const httpClient = Toolbelt.HTTPClient();
+
+  try {
+    const response = await httpClient(process.env['CUSTOMER_URL'], {
+      method: "GET", 
+      headers: {}, 
+      simple: false,
+      resolveWithFullResponse: true
+    });
+      
+    const { statusCode, body } = response;
+    if(statusCode === 200) {
+      callback(null, process.env['SUCCESS_MESSAGE']);
+    } else {
+      callback(null, process.env['ERROR_MESSAGE']);
+    }
+  } catch (err) {
+    callback(err, null)
+  }
+}
+```
+
+See [Environment Variables](liveperson-functions-development-overview.html#environment-variables) for more information about it. 
+
+### Error handling in async functions
+
+If an async call has no correct error handling implemented the function will not end in the execution time and the error `Lambda Execution is taking too long` will be thrown.
```

Comment:
```suggestion
If an async call inside of a LivePerson function has no proper error handling, it could be executed longer than the maximum execution time. In such a case a `Lambda Execution is taking too long` error will be thrown. Therefore proper error handling best practices as the following are vital.
```